extensions/azurecore/src/account-provider/simpleTokenCache.ts

```diff
@@ -0,0 +1,187 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as keytarType from 'keytar';
+import { join, parse } from 'path';
+import { FileDatabase } from './utils/fileDatabase';
+import * as crypto from 'crypto';
+import * as azdata from 'azdata';
+
+function getSystemKeytar(): Keytar | undefined {
+	try {
+		return require('keytar');
```

Comment:
what do we do on Linux systems that don't have an installed key manager?

Reply:
It'll fallback to encrypted file based storage.